[agl] adapt to token logic and nss-localuser

jobol opened the pull request:
Fallback to use the environment variable CYNAGOAUTH_TOKEN
if defined and if no token was provided in the URL.

Signed-off-by: José Bollo <jose.bollo@iot.bzh>

jobol committed:
[agl] Retrieve token from environment

Fallback to use the environment variable CYNAGOAUTH_TOKEN
if defined and if no token was provided in the URL.

Bug-AGL: SPEC-2968
Bug-AGL: SPEC-3032

Signed-off-by: José Bollo <jose.bollo@iot.bzh>

claassistantio commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/webosose/wam?pullRequest=26) <br/>All committers have signed the CLA.

jobol commented:
not compiled and not tested be carefull!!!!

jobol committed:
[agl] Handle hostname of applications

In order to separate applications for cookies and
private data the framework uses the component nss-localuser
to forge specific hostnames.

This commit takes it into account.

Bug-AGL: SPEC-3014

Signed-off-by: José Bollo <jose.bollo@iot.bzh>

jobol commented:
in test

jaragunde commented:
I've been testing too, thanks. Patches work as intended by it looks we need something else to make this integration work.

We have created a branch for icefish called `@6.agl.icefish`. Would you mind to re-submit these patches to that branch? Sorry for the hassle.

jobol commented:
no trouble
